ft-logic/src/lib.rs

```diff
@@ -332,10 +332,13 @@ impl FTLogic {
         self.storage_code_hash = storage_code_hash;
     }
 
+    fn get_id_for_account(account: &ActorId) -> usize{
+        (account.as_ref()[0] / 16) as usize
+    }
+
     fn get_storage_address(&mut self, address: &ActorId) -> ActorId {
-        let encoded = hex::encode(address.as_ref());
-        let id: String = encoded.chars().next().expect("Can't be None").to_string();
-        if let Some(address) = self.id_to_storage.get(&id) {
+        let id = Self::get_id_for_account(address);
+        if let Some(ref address) = self.id_to_storage[id] {
             *address
```

Comment:
No need capture a reference then dereference it.
```suggestion
        if let Some(address) = self.id_to_storage[id] {
            address
```